### PR TITLE
Test operator code generation is up to date

### PR DESCRIPTION
**Pull request by katexochen**
Signed-off-by: Paul Meyer <49727155+katexochen@users.noreply.github.com>

### Proposed change(s)

- This new CI workflow checks that the generated code in the Constellation node operator is up to date.

### CODE REVIEW

File: .github/workflows/test-operator-codegen.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - "release/**"
+    paths:
+      - "operator/constellation-node-operator/**"
+  pull_request:
+    paths:
+      - "operator/constellation-node-operator/**"
```
Comment (malt3) [inline .github/workflows/test-operator-codegen.yml:14]:
```suggestion
      - "operators/constellation-node-operator/**"
  pull_request:
    paths:
      - "operators/constellation-node-operator/**"
```

File: .github/workflows/test-operator-codegen.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Setup Go environment
+        uses: actions/setup-go@c4a742cab115ed795e34d4513e2cf7d472deb55f # tag=v3.3.1
+        with:
+          go-version: "1.19.2"
+          cache: true
+
+      - name: Run code generation
+        shell: bash
+        working-directory: "operators/constellation-node-operator"
+        run: make manifests generate
```
Comment (malt3) [inline .github/workflows/test-operator-codegen.yml:35]: We could abstract this for every folder in `operators/` (for future expansion). But tbh this is simpler so feel free to leave as it is.

Comment (katexochen) [reply .github/workflows/test-operator-codegen.yml:35]: Are we sure that the exact same Makefile will exist for every additional opperator?

Comment (malt3) [reply .github/workflows/test-operator-codegen.yml:35]: If we continue to use kubebuilder (which we really should) then yes.

**Review by malt3 — APPROVED** [review]
LGTM

**Pull request merged**